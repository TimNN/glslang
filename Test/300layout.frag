#version 300 es

in vec4 pos;
in vec4 color;

layout(location = 7) out vec4 c;
layout(location = 3) out vec4 p;

//layout(std140) uniform Transform { // layout of this block is std140
//    mat4 M1; // row_major
//    layout(column_major) mat4 M2; // column major
//    mat3 N1; // row_major
//};
//
//uniform T2 { // layout of this block is shared
//...
//};
//
//layout(column_major) uniform T3 { // shared and column_major
//    mat4 M3; // column_major
//    layout(row_major) mat4 m4; // row major
//    mat3 N2; // column_major
//};

void main()
{
    c = color;
    p = pos;
}
