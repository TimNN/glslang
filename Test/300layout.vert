#version 300 es

layout(location = 7) in vec4 c;
layout(LocatioN = 3) in vec4 p;
out vec4 pos;
out vec4 color;

layout(shared, column_major, row_major) uniform mat4 m4; // default is now shared and row_major

//layout(std140) uniform Transform { // layout of this block is std140
//    mat4 M1; // row_major
//    layout(column_major) mat4 M2; // column major
//    mat3 N1; // row_major
//};
//
//uniform T2 { // layout of this block is shared
//...
//};
//
//layout(column_major) uniform T3 { // shared and column_major
//    mat4 M3; // column_major
//    layout(row_major) mat4 m4; // row major
//    mat3 N2; // column_major
//};

void main()
{
    pos = p * m4;
    color = c;
}
