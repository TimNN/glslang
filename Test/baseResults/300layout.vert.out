ERROR: 0:7: 'vertex input arrays' : not supported with this profile: es
ERROR: 0:8: 'in' : cannot be a structure or array 
ERROR: 0:8: 'vertex input arrays' : not supported with this profile: es
ERROR: 0:12: 'badm4' : cannot specify matrix layout on a variable declaration 
ERROR: 0:12: 'badm4' : cannot specify packing on a variable declaration 
ERROR: 0:15: 'badf' : member of uniform block cannot have an auxiliary or interpolation qualifier 
ERROR: 0:15: 'badg' : member storage qualifier cannot contradict block storage qualifier 
ERROR: 0:28: 'T3' : nameless block contains a member that already has a name at global scope 
ERROR: 0:35: 'badout' : only uniform interface blocks are supported 
ERROR: 0:39: 'badout' : can only use location layout qualifier on a vertex input or fragment output 
ERROR: 0:47: 'shared' : not supported with this profile: es
ERROR: 0:47: 'shared' : not supported in this stage: vertex
ERROR: 12 compilation errors.  No code generated.

ERROR: node is still EOpNull!
0:41  Function Definition: main( (void)
0:41    Function Parameters: 
0:43    Sequence
0:43      move second child to first child (highp 4-component vector of float)
0:43        'pos' (smooth out highp 4-component vector of float)
0:43        vector-times-matrix (highp 4-component vector of float)
0:43          'p' (layout(location=3 ) in highp 4-component vector of float)
0:43          add (highp 4X4 matrix of float)
0:43            add (highp 4X4 matrix of float)
0:43              add (highp 4X4 matrix of float)
0:43                add (highp 4X4 matrix of float)
0:43                  M1: direct index for structure (layout(row_major std140 ) highp 4X4 matrix of float)
0:43                    'tblock' (layout(std140 ) uniform block)
0:43                    0 (const int)
0:43                  M2: direct index for structure (layout(column_major std140 ) highp 4X4 matrix of float)
0:43                    'tblock' (layout(std140 ) uniform block)
0:43                    1 (const int)
0:43                M4: direct index for structure (layout(row_major shared ) highp 4X4 matrix of float)
0:43                  '__anon__1' (layout(shared ) uniform block)
0:43                  1 (const uint)
0:43              M3: direct index for structure (layout(column_major shared ) highp 4X4 matrix of float)
0:43                '__anon__1' (layout(shared ) uniform block)
0:43                0 (const uint)
0:43            t2m: direct index for structure (layout(row_major shared ) highp 4X4 matrix of float)
0:43              '__anon__0' (layout(shared ) uniform block)
0:43              1 (const uint)
0:44      move second child to first child (highp 3-component vector of float)
0:44        'color' (smooth out highp 3-component vector of float)
0:44        vector-times-matrix (highp 3-component vector of float)
0:44          'c' (layout(location=7 ) in highp 3-component vector of float)
0:44          N1: direct index for structure (layout(row_major std140 ) highp 3X3 matrix of float)
0:44            'tblock' (layout(std140 ) uniform block)
0:44            2 (const int)
0:?   Linker Objects
0:?     'c' (layout(location=7 ) in highp 3-component vector of float)
0:?     'p' (layout(location=3 ) in highp 4-component vector of float)
0:?     'q' (layout(location=9 ) in 4-element array of highp 4-component vector of float)
0:?     'r' (layout(location=10 ) in 4-element array of structure)
0:?     'pos' (smooth out highp 4-component vector of float)
0:?     'color' (smooth out highp 3-component vector of float)
0:?     'badm4' (layout(column_major shared ) uniform highp 4X4 matrix of float)
0:?     'tblock' (layout(std140 ) uniform block)
0:?     '__anon__0' (layout(shared ) uniform block)
0:?     'badout' (layout(location=10 ) smooth out highp 4-component vector of float)
0:?     'compute_only' (shared highp 4-component vector of float)
0:?     'gl_VertexID' (gl_VertexId highp int)
0:?     'gl_InstanceID' (gl_InstanceId highp int)

